### ISSUE TITLE
Support different regional variants of English

### ISSUE THREAD

**Issue by harrislapiroff** [mastodon/mastodon#8163]
Currently there's only a single localization for English which features a mix of British and US English (e.g., I see "favourite" 🇬🇧 all over the place, but also "authorize" 🇺🇸). We should support both! And maybe other regional variants too? I'm not really an expert, here.

**Comment by Quenty31** [comment]
I'd be for the all in British as US English is only used in one country and we are taught British English at school :)

**Comment by nightpool** [comment]
what would be the benefit of supporting both varients? they're easily
mutually intelligible

**Comment by harrislapiroff** [comment]
@nightpool That's definitely true. I think, as with many UX touches, it's not just about whether software can practically be used but whether it feels like a smooth experience that users experience as thoughtful, designed for them, trustworthy, and enjoyable. NNG has a 2008 article about English variants here: https://www.nngroup.com/articles/american-vs-british-english-for-web/ It would also have the benefit of feature parity with most other social media apps.

**Comment by holaymolay** [comment]
@Quenty31, according to Wikipedia (https://en.m.wikipedia.org/wiki/English-speaking_world)

Approximately 330 to 360 million people speak English as their first language.

**The United States** has the most native speakers at 258 million. 

Additionally, there are 60 million native English speakers in the United Kingdom, 19 million in Canada, 16.5 million in Australia, 4.8 million in Ireland, and 4.8 million in New Zealand. Other countries also use English as their primary and official languages.

**Comment by Quenty31** [comment]
You would suggest some French from Africa as a model just because they are more over there? Whereas French from France is the easiest to be understood.

Just saying....

**Comment by holaymolay** [comment]
Hardly the same thing when +70% of the native English-speaking world is use en-us. However, its obvious that both should be supported because each dialect is distinct enough to warrant the effort. English, however, is an evolving language -- in spelling as well as pronunciation. An example of this is how 
> non-rhotic (current British pronunciation) speech took off in southern England, around the turn of the 18th 19th century, especially among the upper and upper-middle classes. It was a signifier of class and status. This posh accent was standardized as Received Pronunciation and taught widely by pronunciation tutors to people who wanted to learn to speak fashionably. Because the Received Pronunciation accent was regionally "neutral" and easy to understand, it spread across England and the empire through the armed forces, the civil service and, later, the BBC. 

This is all to say that it is difficult to assert that one is more correct than the other. 

It's easier to say that France's French is more correct -- even though there are different dialects -- because the French people are more diligent about defining an official French vocabulary. English is much more bastardized, regardless of which side of the pond you're on.

**Comment by ykzts** [comment]
You can add British English from https://weblate.joinmastodon.org/projects/mastodon/backend/en_GB/ :tada:

**Comment by LoganDark** [comment]
I second this. "Favorite" is misspelled on my Mastodon instance. Even just at a glance I already saw something out of place. I was about to open a bug report for this, but it seems like there was already an issue for this (and it honestly seems like people only support British English).

I'm honestly offended by the lack of support for proper English. You can argue all you want about which one's better, but the core issue here is that you only offer British English as an option (even the API endpoints are spelt that way, which is honestly ridiculous). As my Mastodon instance is running for a U.S.-based company, it would be nice to have localized text on the network, especially for things that are visible in many places (e.g. Favorites).

Basically, in summary, it's just a little ironic that you offer options for tons of other languages... but not a simple variant of English ("favourite" -> favorite). Even my spell checker finds "favourite" to be wrong.

**Comment by holaymolay** [comment]
@LoganDark while I agree with your point, being "offended" is a bit too strong of an emotion to warrant its invocation. I'm guessing your use is a result of the word being over-used in recent years resulting in a dilution of meaning. At best, the en-uk "snobs" are mildly insensitive, but as long as no ad-hominem attacks are exchanged, all is fair if everyone sticks to arguing the facts.

With that said, the main argument here is over _economy of effort_. Is it worth the time and effort for the mastadon contributing community to put towards the cause?

I guess I'd say Yes, it should be done eventually. How soon is another matter entirely. However, if someone wants to contribute their time and spearhead the effort then I see no harm in that. Isn't that kind of the beauty of open-source? You get to contribute how and when you choose.

**Comment by LoganDark** [comment]
`s/([Ff])([Aa])([Vv])([Oo])([Uu])([Rr])([Ii])([Tt])([Ee])/\1\2\3\4\6\7\8\9/g`

I'm sorry for having a strong reaction to this, but I have seen british spelling used in places where it should not be and it honestly pisses me off. I get fed up of some things easily.

**Comment by Quenty31** [comment]
It should be called English and English US as only a country decided that have a different spelling for weird reasons.

**Comment by LoganDark** [comment]
...I would recommend English UK and English US, to avoid being unfair/

**Comment by holaymolay** [comment]
ok ok, how about we call them "Kinglish" and "EnglUSh" respectively?? :laughing:

**Comment by Quenty31** [comment]
Bah non Logan, British English is used everywhere but in the US, it is not that "British" then, just English :)

**Comment by LoganDark** [comment]
There's British and American english variants though. Not my fault most of the world uses British English.

**Comment by rugk** [comment]
Indeed, can we stop this silly discussion? It's clear what this issue is about and how it is implemented. Naming the language selection is a totally minor (and, actually solved) detail.

**Comment by LoganDark** [comment]
Exactly...

**Comment by stevenroose** [comment]
I hope this is meant for the interface and not for the content filter of posts?

**Comment by Pizzacus** [comment]
**Mastodon is written in American English!** Some of the time…

![image](https://user-images.githubusercontent.com/7659613/63227892-e120a880-c1eb-11e9-8a1f-8ab6dc0ca015.png)

Also, the word "favourite" is actually used **within the source code itself.** Although I personally use British English, I don't think code is supposed to favour it… It creates a lot of confusion if you have to wonder "was this property 'color' or 'colour'?"

I think we can all agree that making a "milkshake" of both dialects like that is unacceptable as everyone ends up being unhappy…

**Comment by Quenty31** [comment]
What's the legitimacy of the US way of spelling words?
Thanks

**Comment by Pizzacus** [comment]
Sorry does a dialect used by millions need to be "legitimate" to be supported? What is this elitism?

If it was me, I'd even support Canadian English, which uses -our and -ize

**Comment by LoganDark** [comment]
@Pizzacus Canadian English is just British English. You could go the phpBB route and support "British English" and "American English" options.

**Comment by Pizzacus** [comment]
@LoganDark no as I said they use the British U like in "labour" and the American Z like in "minimize"

That is indeed pretty minor but it can pretty much be handled automatically with the right setup, so I'd rather give people options ^-^

Not saying Mastodon should do the same, but at least we can agree having the choice between US and UK spellings is a minimum.

**Comment by LoganDark** [comment]
Oh, sorry, I guess I missed that...

British English, American English, Canadian English then?

**Comment by Pizzacus** [comment]
Well, it's up to you guys for Canadian, it depends on how messy having 3 English locales would be.

**Comment by LoganDark** [comment]
Well you could always go the route of most of the world and only support British English and Windows

**Comment by ryliejamesthomas** [comment]
I would still like to see this done.

**Comment by kueda** [comment]
@Gargron, this has been open since 2018. If you're using Crowdin in the way I've used it, this should be a matter of enabling `en-US` as an option there, right? Personally I'm in favor of enabling it for the benefit of instances concerning people in the US. If you or the other maintainers have a reason for not supporting US English localization, would you please state it and close this issue?

**Comment by tbjers** [comment]
What I found the most amusing about this entire thread was the one French account trolling everyone about what form of English is the best and most accepted while also trying to convince people that French from France is better than French from any other place. How very French of them.

**Comment by LoganDark** [comment]
> What I found the most amusing about this entire thread was the one French account trolling everyone about what form of English is the best and most accepted while also trying to convince people that French from France is better than French from any other place. How very French of them.

I'm just shocked at the comments I left here back in 2018. This whole thread is a mess for all sorts of reasons.

**Comment by tbjers** [comment]
> I'm just shocked at the comments I left here back in 2018. This whole thread is a mess for all sorts of reasons.

For sure. I came here looking to see if anyone had tackled the i18n issue with words and verbiage being comingled between British and US English words.

**Comment by mririgoyen** [comment]
Now that Mastodon is seeing a high influx on Twitter refugees, especially in the States, I feel that it might be time to seriously consider getting `en-US` working. As a new Mastodon user, see "favourites" immediately triggered me that something was awry.

It's also incorrect in my spellchecker. 🤣 
![Screenshot 2022-11-19 at 7 52 22 PM](https://user-images.githubusercontent.com/1015040/202878958-71d5b119-e842-4a07-9456-5dd779c9d1d0.png)

**Comment by gunchleoc** [comment]
It's common practice to have US English in the source strings, and British English as a translation project, like this: https://crowdin.com/project/mastodon/en-GB

The difficulty is that the main developer will have had British orthography drummed into their head at school.

I have made a PR to fix the "favo(u)rite" issue, are there any more words that I should look at?

**Comment by renchap** [comment]
Closing this issue as the UI supports both `en-US` and `en-UK`. Please open individual issues (or better, pull requests) if some translated strings in the codebase are not using `en-US`.
